authentik: init at 2023.10.7

jvanbruegge opened the pull request:
## Description of changes

This packages the authentik IdP. I've verified that it works, I was able to use the LDAP outpost to log into tandoor-recipes (also from nixpkgs).

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jvanbruegge committed:
authentik: init at 2023.10.7

jvanbruegge committed:
authentik-outposts.ldap: init at 2023.10.7

bjornfor reviewed (commented):
New packages go in pkgs/by-name whenever possible. See https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md.

jvanbruegge commented:
done

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/3085

AndersonTorres commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,226 @@
+{ lib, stdenvNoCC, fetchFromGitHub, buildNpmPackage, buildGoModule, runCommand, openapi-generator-cli, nodePackages, poetry, python3, codespell, makeWrapper }:
```
One per line

jvanbruegge replied:
done

AndersonTorres commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,226 @@
+      xmlsec
+      zxcvbn
+      jsonpatch
+    ] ++ [
```
```suggestion
```

jvanbruegge replied:
This is necessary because codespell is also in `python3.pkgs` but throws an error saying it moved to the toplevel

AndersonTorres commented on pkgs/tools/networking/openapi-generator-cli/default.nix:
```diff
@@ -33,6 +33,7 @@ let this = stdenv.mkDerivation rec {
     homepage = "https://github.com/OpenAPITools/openapi-generator";
     changelog = "https://github.com/OpenAPITools/openapi-generator/releases/tag/v${version}";
     sourceProvenance = with sourceTypes; [ binaryBytecode ];
+    mainProgram = pname;
```
```suggestion
    mainProgram = "openapi-generator-cli";
```

AndersonTorres commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,226 @@
+    owner = "goauthentik";
+    repo = "authentik";
+    rev = "version/${version}";
+    sha256 = "sha256-NsXYfXxn7ofl9EeLAQi1V0tKlOeVyQqH1W26uejjKV0=";
```
```suggestion
    hash = "sha256-NsXYfXxn7ofl9EeLAQi1V0tKlOeVyQqH1W26uejjKV0=";
```

jvanbruegge commented:
done

SuperSandro2000 commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,236 @@
+  meta = with lib; {
+    description = " The authentication glue you need. ";
+    changelog = "https://github.com/goauthentik/authentik/releases/tag/version%2F${version}";
+    homepage = "goauthentik.io";
```
```suggestion
    homepage = "https://goauthentik.io/";
```

SuperSandro2000 commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,236 @@
+  nativeBuildInputs = [ makeWrapper ];
+
+  meta = with lib; {
+    description = " The authentication glue you need. ";
```
```suggestion
    description = "The authentication glue you need";
```

SuperSandro2000 commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,236 @@
+    mkdir -p $out/bin
+    cp -r lifecycle/* $out/bin/
+
+    wrapProgram $out/bin/ak --prefix PATH : ${
+      lib.makeBinPath [ (python3.withPackages (_: [authentik-django])) proxy ]
+    } --set TMPDIR /dev/shm \
```
```suggestion
    wrapProgram $out/bin/ak \
      --prefix PATH : ${lib.makeBinPath [ (python3.withPackages (ps: [ps.authentik-django])) proxy ]} \
      --set TMPDIR /dev/shm \
```
we don't want to accidentally a wrong python interpreter. Also setting TMPDIR to /dev/shm seems strange.

jvanbruegge replied:
The TMPDIR is taken from upstream

jvanbruegge replied:
Also the package is the one that is build earlier, it is not part of the python package set. And it cannot be out of sync because it is built from the same python as this

jvanbruegge replied:
how would I do that?

SuperSandro2000 replied:
I had something like this https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/web-apps/healthchecks/default.nix#L8-L12 in mind

SuperSandro2000 commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,236 @@
+  pname = "authentik";
+  inherit src version;
+
+  patchPhase = ''
```
```suggestion
  postPatch = ''
```
we don't want to break patches

SuperSandro2000 commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,236 @@
+      cp ${src}/go.mod ${src}/go.sum $out/
+    '';
+
+    patchPhase = ''
```
```suggestion
    postPatch = ''
```

SuperSandro2000 commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,236 @@
+    inherit version;
+
+    src = runCommand "authentik-proxy-source" {} ''
+      mkdir -p $out/web
+      mkdir -p $out/authentik
```
```suggestion
      mkdir -p $out/web $out/authentik
```

SuperSandro2000 commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,236 @@
+    inherit version src;
+    format = "pyproject";
+
+    patchPhase = ''
```
```suggestion
    postPatch = ''
```

SuperSandro2000 commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,236 @@
+  authentik-django = python3.pkgs.buildPythonPackage {
+    pname = "authentik-django";
+    inherit version src;
+    format = "pyproject";
```
```suggestion
    pyproject = true;
```

SuperSandro2000 commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,236 @@
+    '';
+    npmDepsHash = "sha256-5aCKlArtoEijGqeYiY3zoV0Qo7/Xt5hSXbmy2uYZpok=";
+
+    postPatch = "cd web";
```
```suggestion
    preBuild = ''
      cd web
    '';
```

jvanbruegge replied:
Doesn't work, `buildNpmPackage` does not find the package-lock.json otherwise.

SuperSandro2000 commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,236 @@
+    inherit version;
+    src = runCommand "authentik-webui-source" {} ''
+      mkdir -p $out/web/node_modules/@goauthentik/
+      cp -r ${src}/web $out/
+      cp -r ${src}/website $out/
```
Using symlinks is not possible when constructing the src?

jvanbruegge replied:
The first one has to be copied because the build tried to write in there, I've symlinked the rest

SuperSandro2000 commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,236 @@
+    pname = "authentik-client-api";
+    inherit version src;
+
+    patchPhase = ''
```
```suggestion
    postPatch = ''
```

SuperSandro2000 commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,236 @@
+      openapi-generator-cli generate -i ./schema.yml \
+      -g typescript-fetch -o $out \
+      -c ./scripts/api-ts-config.yaml \
+      --additional-properties=npmVersion=${nodePackages.npm.version} \
```
This is not the npm used in nodejs and buildNpmPackage

SuperSandro2000 commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,236 @@
+    NODE_ENV = "production";
+    NODE_OPTIONS = "--openssl-legacy-provider";
+
+    postPatch = "cd website";
+
+    installPhase = "cp -r help $out";
```
```suggestion
    preBuild = ''
      cd website
    '';

    installPhase = ''
      cp -r help $out
    '';
```

SuperSandro2000 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -3377,6 +3377,8 @@ with pkgs;
 
   authelia = callPackage ../servers/authelia { };
 
+  authentik-outposts = recurseIntoAttrs (callPackage ../by-name/au/authentik/outposts.nix { });
```
```suggestion
  authentik-outposts = recurseIntoAttrs (callPackages ../by-name/au/authentik/outposts.nix { });
```
otherwise overrides don't work

jvanbruegge commented:
done

SuperSandro2000 reviewed (approved):
other than the python thingy, LGTM

jvanbruegge commented:
I think I have fixed the python thing, all should be good now

pbsds commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,248 @@
+        --replace '/local' "$(pwd)/"
+    '';
+
+    nativeBuildInputs = [ openapi-generator-cli ];
+    buildPhase = ''
+      openapi-generator-cli generate -i ./schema.yml \
+      -g typescript-fetch -o $out \
+      -c ./scripts/api-ts-config.yaml \
+        --additional-properties=npmVersion=${nodejs.pkgs.npm.version} \
+      --git-repo-id authentik --git-user-id goauthentik
+    '';
```
```suggestion
    buildPhase = ''
      runHook preBuild
      openapi-generator-cli generate -i ./schema.yml \
      -g typescript-fetch -o $out \
      -c ./scripts/api-ts-config.yaml \
        --additional-properties=npmVersion=${nodejs.pkgs.npm.version} \
      --git-repo-id authentik --git-user-id goauthentik
      runHook postBuild
    '';
```

AndersonTorres replied:
```suggestion
    nativeBuildInputs = [ openapi-generator-cli ];
    
    buildPhase = ''
      runHook preBuild
      
      openapi-generator-cli generate \
        -i ./schema.yml \
        -g typescript-fetch \
        -o $out \
        -c ./scripts/api-ts-config.yaml \
        --additional-properties=npmVersion=${nodejs.pkgs.npm.version} \
        --git-repo-id authentik \
        --git-user-id goauthentik
        
      runHook postBuild
    '';
```

pbsds commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,248 @@
+      cd web
+    '';
+
+    installPhase = ''
+      mkdir $out
+      cp -r dist $out/dist
+      cp -r authentik $out/authentik
+    '';
```
```suggestion
    installPhase = ''
      runHook preInstall
      mkdir $out
      cp -r dist $out/dist
      cp -r authentik $out/authentik
      runHook postInstall
    '';
```

pbsds commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,248 @@
+      --replace '> /dev/stderr' ""
+  '';
+
+  installPhase = ''
```
```suggestion
  installPhase = ''
    runHook preInstall
```

pbsds commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,248 @@
+      --set TMPDIR /dev/shm \
+      --set PYTHONDONTWRITEBYTECODE 1 \
+      --set PYTHONUNBUFFERED 1
+  '';
```
```suggestion
    runHook postInstall
  '';
```

pbsds commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,248 @@
+    npmInstallFlags = [ "--include=dev" ];
+  };
+
+  python = python3.override {
```
```suggestion
  python = python3.override {
    self = python;
```

pbsds replied:
With this change i believe many of the `final` below can be turned into `prev`

pbsds commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,248 @@
+    };
+  };
+
+  authentik-django = python.pkgs.authentik-django;
```
```suggestion
  inherit (python.pkgs) authentik-django;
```

pbsds commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,248 @@
+
+      cp -r ${src}/cmd ${src}/internal $out/
+      cp -r ${src}/authentik/lib $out/authentik/
+      cp ${src}/web/static.go $out/web/
+
+      cp ${src}/web/robots.txt ${src}/web/security.txt $out/web/
```
```suggestion
      cp ${src}/web/* $out/web/
```

Any reason to not copy everything?

pbsds commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,248 @@
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp -r lifecycle/* $out/bin/
```
It seems the following files are not wrapped nor are their shebang patched.

* results/authentik/bin/migrate.py
* results/authentik/bin/wait_for_db.py

Should they even be in `bin`? Their names are pretty generic.

pbsds reviewed (commented):
Result of `nixpkgs-review pr 271885` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>authentik</li>
    <li>authentik-outposts.ldap</li>
  </ul>
</details>

AndersonTorres commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,248 @@
+
+  webui = buildNpmPackage {
+    pname = "authentik-webui";
+    inherit version;
+    src = runCommand "authentik-webui-source" {} ''
```
```suggestion
    inherit version;
    
    src = runCommand "authentik-webui-source" {} ''
```

AndersonTorres commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,248 @@
+      cp -r authentik $out/authentik
+    '';
+
+    NODE_ENV = "production";
+    NODE_OPTIONS = "--openssl-legacy-provider";
```
```suggestion
    env = {
      NODE_ENV = "production";
      NODE_OPTIONS = "--openssl-legacy-provider";
    };
```

pbsds replied:
generally i prefer 
```suggestion
    env.NODE_ENV = "production";
    env.NODE_OPTIONS = "--openssl-legacy-provider";
```

which can easily be applied treewide

jvanbruegge commented:
It is already included, see here how I run it: https://github.com/jvanbruegge/lunchbox/blob/master/modules/authentik.nix#L58

jvanbruegge commented:
I have sorted out all review comments now

pbsds commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,249 @@
+    pname = "authentik-proxy";
+    inherit version;
+
+    src = runCommand "authentik-proxy-source" {} ''
+       cp -r ${src} $out
+    '';
```
This should work, and saves a derivation

```suggestion
    inherit src;
```

pbsds commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,249 @@
+    changelog = "https://github.com/goauthentik/authentik/releases/tag/version%2F${version}";
+    homepage = "https://goauthentik.io/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ jvanbruegge ];
```
```suggestion
    maintainers = with maintainers; [ jvanbruegge ];
    mainProgram = "ak";
```

pbsds commented on pkgs/by-name/au/authentik/ldap.nix:
```diff
@@ -0,0 +1,12 @@
+
+buildGoModule {
+  pname = "authentik-ldap-outpost";
+  inherit (authentik) version src meta;
```
```suggestion
  inherit (authentik) version src;
```

pbsds commented on pkgs/by-name/au/authentik/ldap.nix:
```diff
@@ -0,0 +1,12 @@
+  CGO_ENABLED = 0;
+
+  subPackages = [ "cmd/ldap" ];
+}
```
```suggestion

  meta = authentik.meta // {
    mainProgram = "ldap";
  };
}
```

pbsds replied:
Perhaps a few other changes as well, explaining what this outpost is and does

pbsds reviewed (approved):
Sorry for being unresponsive during the holidays.

Result of `nixpkgs-review pr 271885` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>authentik</li>
    <li>authentik-outposts.ldap</li>
  </ul>
</details>


- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

Some final nits, then LGTM

jvanbruegge commented:
@pbsds  I've addressed the review comments, and I also updated to the latest release. I verified that this still works as expected (including the ldap provider)

jvanbruegge commented:
@ofborg build authentik

pbsds commented:
From ofborg:
```
Checking runtime dependencies for authentik-2023.10.6-py3-none-any.whl
  - djangorestframework-guardian not installed
  - dumb-init not installed
```

Of note, recently python3.12 has become the default, and dependency version checking is re-enabled

jvanbruegge commented:
Ok, this should be fixed now. But very annoying that pythonRelaxDepsHook doesnt work any more with the new runtime deps check

jvanbruegge commented:
Yesterday a new release was cut, so I've also updated the version here

SuperSandro2000 commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,250 @@
+    hash = "sha256-+1IdXRt28UZ2KTa0zsmjneNUOcutP99UUwqcYyVyqTI=";
+  };
+
+  website = buildNpmPackage {
```
The intermediate packages are missing meta

SuperSandro2000 commented on pkgs/by-name/au/authentik/package.nix:
```diff
@@ -0,0 +1,250 @@
+    NODE_ENV = "production";
+    NODE_OPTIONS = "--openssl-legacy-provider";
+
+    npmInstallFlags = [ "--include=dev" ];
```
which is relevant here because darwin fails to build and I think we should just mark it linux only or broken on darwin.

```
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
npm ERR! code 1
npm ERR! path /private/tmp/nix-build-authentik-webui-2023.10.7.drv-0/authentik-webui-source/web/node_modules/tree-sitter-yaml
npm ERR! command failed
npm ERR! command sh -c node-gyp rebuild
npm ERR! make: Entering directory '/private/tmp/nix-build-authentik-webui-2023.10.7.drv-0/authentik-webui-source/web/node_modules/tree-sitter-yaml/build'
npm ERR!   CXX(target) Release/obj.target/tree_sitter_yaml_binding/bindings/node/binding.o
npm ERR! make: Leaving directory '/private/tmp/nix-build-authentik-webui-2023.10.7.drv-0/authentik-webui-source/web/node_modules/tree-sitter-yaml/build'
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@10.0.1
npm ERR! gyp info using node@20.11.0 | darwin | x64
npm ERR! gyp info find Python using Python version 3.11.7 found at "/nix/store/byyb569c90slwsjr43ikhhwlhfavhiir-python3-3.11.7/bin/python3"
npm ERR! gyp WARN read config.gypi ENOENT: no such file or directory, open '/nix/store/kiprv8l36cv4c68bqpd2hcl17py4400w-nodejs-20.11.0-source/include/node/config.gypi'
npm ERR! gyp info spawn /nix/store/byyb569c90slwsjr43ikhhwlhfavhiir-python3-3.11.7/bin/python3
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args '/nix/store/zgmy5yc85d3mkz6zdmazv799c45f2gkh-nodejs-20.11.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args 'binding.gyp',
npm ERR! gyp info spawn args '-f',
npm ERR! gyp info spawn args 'make',
npm ERR! gyp info spawn args '-I',
npm ERR! gyp info spawn args '/private/tmp/nix-build-authentik-webui-2023.10.7.drv-0/authentik-webui-source/web/node_modules/tree-sitter-yaml/build/config.gypi',
npm ERR! gyp info spawn args '-I',
npm ERR! gyp info spawn args '/nix/store/zgmy5yc85d3mkz6zdmazv799c45f2gkh-nodejs-20.11.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args '-I',
npm ERR! gyp info spawn args '/nix/store/kiprv8l36cv4c68bqpd2hcl17py4400w-nodejs-20.11.0-source/common.gypi',
npm ERR! gyp info spawn args '-Dlibrary=shared_library',
npm ERR! gyp info spawn args '-Dvisibility=default',
npm ERR! gyp info spawn args '-Dnode_root_dir=/nix/store/kiprv8l36cv4c68bqpd2hcl17py4400w-nodejs-20.11.0-source',
npm ERR! gyp info spawn args '-Dnode_gyp_dir=/nix/store/zgmy5yc85d3mkz6zdmazv799c45f2gkh-nodejs-20.11.0/lib/node_modules/npm/node_modules/node-gyp',
npm ERR! gyp info spawn args '-Dnode_lib_file=/nix/store/kiprv8l36cv4c68bqpd2hcl17py4400w-nodejs-20.11.0-source/$(Configuration)/node.lib',
npm ERR! gyp info spawn args '-Dmodule_root_dir=/private/tmp/nix-build-authentik-webui-2023.10.7.drv-0/authentik-webui-source/web/node_modules/tree-sitter-yaml',
npm ERR! gyp info spawn args '-Dnode_engine=v8',
npm ERR! gyp info spawn args '--depth=.',
npm ERR! gyp info spawn args '--no-parallel',
npm ERR! gyp info spawn args '--generator-output',
npm ERR! gyp info spawn args 'build',
npm ERR! gyp info spawn args '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp info spawn make
npm ERR! gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm ERR! In file included from ../bindings/node/binding.cc:2:
npm ERR! In file included from /nix/store/kiprv8l36cv4c68bqpd2hcl17py4400w-nodejs-20.11.0-source/src/node.h:73:
npm ERR! In file included from /nix/store/kiprv8l36cv4c68bqpd2hcl17py4400w-nodejs-20.11.0-source/deps/v8/include/v8.h:21:
npm ERR! In file included from /nix/store/cxzbqsaziwy6n2p9qz88l2vx5yns4h9n-libcxx-16.0.6-dev/include/c++/v1/memory:884:
npm ERR! In file included from /nix/store/cxzbqsaziwy6n2p9qz88l2vx5yns4h9n-libcxx-16.0.6-dev/include/c++/v1/__memory/allocate_at_least.h:13:
npm ERR! In file included from /nix/store/cxzbqsaziwy6n2p9qz88l2vx5yns4h9n-libcxx-16.0.6-dev/include/c++/v1/__memory/allocator_traits.h:14:
npm ERR! In file included from /nix/store/cxzbqsaziwy6n2p9qz88l2vx5yns4h9n-libcxx-16.0.6-dev/include/c++/v1/__memory/construct_at.h:23:
npm ERR! /nix/store/cxzbqsaziwy6n2p9qz88l2vx5yns4h9n-libcxx-16.0.6-dev/include/c++/v1/new:355:14: error: no member named 'aligned_alloc' in the global namespace
npm ERR!     return ::aligned_alloc(__alignment, __size > __rounded_size ? __size : __rounded_size);
npm ERR!            ~~^
npm ERR! 1 error generated.
npm ERR! make: *** [tree_sitter_yaml_binding.target.mk:128: Release/obj.target/tree_sitter_yaml_binding/bindings/node/binding.o] Error 1
npm ERR! gyp ERR! build error 
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack at ChildProcess.<anonymous> (/nix/store/zgmy5yc85d3mkz6zdmazv799c45f2gkh-nodejs-20.11.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:209:23)
npm ERR! gyp ERR! System Darwin 22.2.0
npm ERR! gyp ERR! command "/nix/store/zgmy5yc85d3mkz6zdmazv799c45f2gkh-nodejs-20.11.0/bin/node" "/nix/store/zgmy5yc85d3mkz6zdmazv799c45f2gkh-nodejs-20.11.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
npm ERR! gyp ERR! cwd /private/tmp/nix-build-authentik-webui-2023.10.7.drv-0/authentik-webui-source/web/node_modules/tree-sitter-yaml
npm ERR! gyp ERR! node -v v20.11.0
npm ERR! gyp ERR! node-gyp -v v10.0.1
npm ERR! gyp ERR! not ok
npm ERR! Log files were not written due to an error writing to the directory: /nix/store/z893q8w83kkz21ikd50y0k7c35ni0brz-authentik-webui-2023.10.7-npm-deps/_logs
npm ERR! You can rerun the command with `--loglevel=verbose` to see the logs in your terminal
error: builder for '/nix/store/0y5p4cmxgcmg50yy9fyannrdb8x6xaff-authentik-webui-2023.10.7.drv' failed with exit code 1;
```

SuperSandro2000 reviewed (commented):
one final step, then we can merge this

jvanbruegge commented:
@SuperSandro2000 done

jvanbruegge commented:
Ok, in the meantime `meta.mainProgram` was already set for openapi-generator-cli on master, dropped that commit now

pbsds commented:
Result of `nixpkgs-review pr 271885` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>authentik</li>
    <li>authentik-outposts.ldap</li>
  </ul>
</details>

ofborg may be misbehaving

@ofborg build authentic authentik-outposts.ldap authentik-outposts.ldap.passthru.tests authentik.passthru.tests

pbsds commented:
The x86_64-linux  ofborg error seems to happen in `python311Packages.celery`, but hydra is green https://hydra.nixos.org/job/nixos/trunk-combined/nixpkgs.python311Packages.celery.x86_64-linux, and my nixpkgs-review on x86_64-linux was good. 
The aarch64-linux ofborg is also good and darwin is not supported.

LGTM

jvanbruegge commented:
Thanks!
